src/cli/mod.rs

```diff
@@ -58,13 +60,38 @@ pub enum Command {
 }
 
 fn completion(args: CompletionCommand) -> miette::Result<()> {
+    let clap_shell = args
+        .shell
+        .or(clap_complete::Shell::from_env())
+        .unwrap_or(clap_complete::Shell::Bash);
     clap_complete::generate(
-        args.shell.or(Shell::from_env()).unwrap_or(Shell::Bash),
+        clap_shell,
         &mut Args::command(),
         "pixi",
         &mut std::io::stdout(),
     );
 
+    // Create PS1 overwrite command
+    let mut script = String::new();
+    let shell = ShellEnum::from_str(clap_shell.to_string().as_str()).into_diagnostic()?;
```

Comment:
We should check if this works for all supported shells

Reply:
It doesn't ill make an issue for that.